src/main/java/com/linkedin/kmf/services/ConsumerFactoryImpl.java

```diff
@@ -34,9 +37,16 @@
       new String[] {ConsumeServiceConfig.BOOTSTRAP_SERVERS_CONFIG, ConsumeServiceConfig.ZOOKEEPER_CONNECT_CONFIG};
   private int _latencySlaMs;
   private static AdminClient adminClient;
+  private static final Logger LOGGER = LoggerFactory.getLogger(ConsumerFactoryImpl.class);
 
   public ConsumerFactoryImpl(Map<String, Object> props) throws Exception {
+    LOGGER.info("Creating AdminClient.");
     adminClient = AdminClient.create(props);
+    try {
```

Comment:
I think it's in general a good idea to test adminClient in the constructor of this class. However, in the catch statement, if the `ExecutionException` exception was thrown, that indicates some issue with the admin client or the Kafka cluster, right? For example,

1. The admin client is not configured properly
2. The Kafka cluster has some ongoing issues so that metadata can not be returned
3. The network connect between admin client and the Kafka cluster has problems

So, is it a good idea to just log the exception and move on? Because we can potentially return a broken admin client to users of this factory class. Can we consider throw the exception in the constructor?

Reply:
Done